lib/vagrant-google/action/read_ssh_info.rb

```diff
@@ -25,12 +25,17 @@ def initialize(app, env)
         end
 
         def call(env)
-          env[:machine_ssh_info] = read_ssh_info(env[:google_compute], env[:machine])
+          # Get the zone we're going to use
+          zone = env[:machine].provider_config.zone
+          # Check if we should use public or private IP address of the instance
+          use_private_ip = env[:machine].provider_config.get_zone_config(zone).use_private_ip
+
+          env[:machine_ssh_info] = read_ssh_info(env[:google_compute], env[:machine], use_private_ip)
 
           @app.call(env)
         end
 
-        def read_ssh_info(google, machine)
+        def read_ssh_info(google, machine, use_private_ip)
```

Comment:
It does make more sense if you have a whole zone set to use the private IP if you have a VPN setup. I'll update the branch along with adding other changes.